torch/csrc/jit/codegen/cuda/lower_predicate.cpp

```diff
@@ -307,6 +436,16 @@ bool PredicateElimination::needsPredicate(Expr* expr) const {
            });
   });
 
+  if (auto uop = dynamic_cast<UnaryOp*>(expr)) {
+    if (uop->getUnaryOpType() == UnaryOpType::LD_MATRIX ||
+        uop->getUnaryOpType() == UnaryOpType::LD_MATRIXT ||
+        uop->getUnaryOpType() == UnaryOpType::CP_ASYNC) {
+      TORCH_INTERNAL_ASSERT(
+          !filters.back()(expr),
+          "No support yet for shared mem primitives with shift");
+    }
+  }
```

Comment:
Why is this done here? It seems this validates the ld_matrix and cp_async are not used with halo-extended smem buffers. The restriction is fine, but should the validation be done in this class?

Reply:
Not needed since assertOnWarpOps should handle it. Removed. Thanks!